server/routes/api/messages.js

```diff
@@ -43,4 +43,22 @@ router.post("/", async (req, res, next) => {
   }
 });
 
+// The following route only accepts bulk 
+// update of the message resource
+router.put("/", async (req, res, next) => {
+  const { read, unreadMessages } = req.body
+  try{
+    const message = await Message.update(
+      { read: read},
+      { where: {
+        id: unreadMessages
+      }}
+    )
```

Comment:
This route should be protected against users not part of a conversation. See the comment above about updating all messages using a conversation ID.

Reply:
Fixed